### PR TITLE
add button

### PR DESCRIPTION
**Pull request by Experiment145**

### CODE REVIEW

File: src/components/table/Table.js [hunk 1]
```diff
@@ -6,22 +6,66 @@ import './Table.css';
...
+    people: [],
+    filterPeople: []
   }
+
   componentWillMount () {
     axios.get("http://jsonplaceholder.typicode.com/users")
     .then(res => {
-      this.setState({ people: res.data });
+      this.setState({ people: res.data,
+                      filterPeople: res.data});
```
Comment (Rudenkopolina) [inline src/components/table/Table.js]: please write it like 
```
this.setState({
...
});
```

File: src/components/table/Table.js [hunk 2]
```diff
@@ -6,22 +6,66 @@ import './Table.css';
...
     })
   }
+  filterBySouthPosition = () => {
+    const aaa = this.state.people.filter(item => item.address.geo.lat > 0);
```
Comment (Rudenkopolina) [inline src/components/table/Table.js]: please, use more suitable name for this variable. Or don't use variable at all, and filter the array in setState like: 
```
this.setState({ 
   filteredPeople: this.state.people.filter...
}); 
```

File: src/components/table/Table.js [hunk 3]
```diff
@@ -6,22 +6,66 @@ import './Table.css';
 class Table extends Component {
 
   state = {
-    people: []
+    people: [],
+    filterPeople: []
```
Comment (Rudenkopolina) [inline src/components/table/Table.js]: please, rename it for filteredPeople

File: src/components/table/Table.js [hunk 4]
```diff
@@ -6,22 +6,66 @@ import './Table.css';
...
+      }
+      return 0;
+    });
+    this.setState({ filterPeople: ddd })
+  }
+
+
   render() {
+
     //  const newArray = this.state.people.filter(item =>
```
Comment (Rudenkopolina) [inline src/components/table/Table.js]: delete this please

File: src/components/table/Table.js [hunk 5]
```diff
@@ -6,22 +6,66 @@ import './Table.css';
...
+          <button className='button' onClick={this.resetData}>
+            Reset
+          </button>
+          <button classNam='button'  onClick={this.sortByName}>
+            Sort by name
+        </button>
+        </div>
         <table className='table' width='100%'>
           <tr>
             <th className='popa__1' width='5.4%'> ID </th>
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: rename this className please

File: src/components/table/Table.js [hunk 6]
```diff
@@ -30,15 +74,15 @@ class Table extends Component {
...
-            .filter((item, index) => index < 5) 
+          {this.state.filterPeople
+            //.filter(item => item.username.length > 7)
+            //.filter((item, index) => index < 5)
             .map(item =>
             <tr>
               <th className='popa__1' width='5.4%'> {item.id} </th>
               <th className='popa__1' width='22.1%'> {item.name} </th>
               <th className='popa__1' width='7%'>
-              {item.username.length} </th>
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: write this like
<th ...>
  ...
</th>

File: src/components/table/Table.js [hunk 7]
```diff
@@ -30,15 +74,15 @@ class Table extends Component {
             <th className='popa__1' width='24.6%'> Email </th>
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: write `width` in `css `file

File: src/components/table/Table.js [hunk 8]
```diff
@@ -30,15 +74,15 @@ class Table extends Component {
...
             <th className='popa__2' width='40.9%'> Adress </th>
           </tr>
-          {this.state.people
-            .filter(item => item.username.length > 7)
-            .filter((item, index) => index < 5) 
+          {this.state.filterPeople
+            //.filter(item => item.username.length > 7)
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: delete this comments please

File: src/components/table/Table.js [hunk 9]
```diff
@@ -6,41 +6,102 @@ import './Table.css';
...
+      this.setState({
+        people: res.data,
+        filteredPeople: res.data
+      });
     })
   }
+  filterBySouthPosition = () => {
+    this.setState({
+      filteredPeople: this.state.people.filter(
+        item => item.address.geo.lat > 0)
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: this round bracket "`)`" should be on the next line

File: src/components/table/Table.js [hunk 10]
```diff
@@ -6,41 +6,102 @@ import './Table.css';
...
+    });
+  }
+  filterByNorthPosition = () => {
+    this.setState({
+      filteredPeople: this.state.people.filter(
+        item => item.address.geo.lat < 0)
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: this round bracket "`)`" should be on the next line

File: src/components/table/Table.js [hunk 11]
```diff
@@ -6,41 +6,102 @@ import './Table.css';
...
+    });
+  }
+    resetData = () => {
```
Comment (IvanZharnov) [inline src/components/table/Table.js]: tabulation!

**Pull request merged**